src/main/java/de/tum/in/www1/artemis/config/migration/setups/localvc/GitLabJenkinsToLocalVCMigration.java

```diff
@@ -0,0 +1,52 @@
+package de.tum.in.www1.artemis.config.migration.setups.localvc;
+
+import static de.tum.in.www1.artemis.config.Constants.PROFILE_MIGRATE_GITLAB_JENKINS_TO_LOCALVC;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.Profile;
+import org.springframework.context.event.EventListener;
+import org.springframework.stereotype.Component;
+
+import de.tum.in.www1.artemis.config.migration.setups.localvc.gitlab.MigrationEntryGitLabToLocalVC;
+import de.tum.in.www1.artemis.config.migration.setups.localvc.jenkins.MigrationEntryJenkinsToLocalVC;
+
+@Component
+@Profile("gitlab & jenkins & " + PROFILE_MIGRATE_GITLAB_JENKINS_TO_LOCALVC)
+public class GitLabJenkinsToLocalVCMigration {
+
+    private static final Logger log = LoggerFactory.getLogger(GitLabJenkinsToLocalVCMigration.class);
+
+    private final ApplicationContext applicationContext;
+
+    private final MigrationEntryGitLabToLocalVC migrationEntryGitLabToLocalVC;
+
+    private final MigrationEntryJenkinsToLocalVC migrationEntryJenkinsToLocalVC;
+
+    public GitLabJenkinsToLocalVCMigration(ApplicationContext applicationContext, MigrationEntryGitLabToLocalVC migrationEntryGitLabToLocalVC,
+            MigrationEntryJenkinsToLocalVC migrationEntryJenkinsToLocalVC) {
+        this.applicationContext = applicationContext;
+        this.migrationEntryGitLabToLocalVC = migrationEntryGitLabToLocalVC;
+        this.migrationEntryJenkinsToLocalVC = migrationEntryJenkinsToLocalVC;
+    }
+
+    /**
+     * Entry point for this migration, which is triggered at the startup of Artemis.
+     * Performs the migration and shutdown Artemis afterward.
+     */
+    @EventListener(ApplicationReadyEvent.class)
+    public void onApplicationInitialized() {
+        log.info("Migration GitLab-Jenkins to LocalVC-Jenkins starting");
+
+        log.info("Migration starts with the Jenkins build plans");
+        migrationEntryJenkinsToLocalVC.execute();
+        log.info("Migration follow ups with copy the repositories and adjusting the urls in the database");
+        migrationEntryGitLabToLocalVC.execute();
+        log.info("Migration GitLab-Jenkins to LocalVC-Jenkins finished (Check previous log if there are errors with some repositories)");
+
+        SpringApplication.exit(applicationContext, () -> 0);
```

Comment:
The result of `SprintApplication.exit()` should be passed to `System.exit()` to prevent the `IllegalStateException` at the end. See [the docs](https://docs.spring.io/spring-boot/reference/features/spring-application.html#features.spring-application.application-exit).

```suggestion
        int exitCode = SpringApplication.exit(applicationContext, () -> 0);
        System.exit(exitCode);
```

Reply:
In case of running the application not as a standalone JAR with embedded Tomcat but on a JavaEE server (Tomcat, Wildfly) this would completely kill the running JVM with all applications on it. Therefore, I don’t think calling `System.exit` is the right thing here.

The documentation mentions that the migration exits with an exception and that the actual status is reported by the previous log message.

Reply:
I see. But I don't see any mention of an exception in the documentation. Only the `Application run failed` message is mentioned. If there is no good way to prevent the exception we should mention it explicitly. I expected the error message, but the exception was throwing me off.

Reply:
I extended the documentation with a800f29e1f6063f47c6a18dd1a4e99e15b19adb6. Now it should be clear, isn't it?

Reply:
That's good, thanks :+1: